File: intervals/summary-ranges/summary-ranges.go
func summaryRanges(nums []int) []string {
    if len(nums) == 0 {
        return nil
    }

	var result []string
    start := nums[0]

    for i := 1; i <= len(nums); i++ {
        if i == len(nums) || nums[i] != nums[i - 1] + 1 {
            end := nums[i - 1]
			if start == end {
				result = append(result, strconv.Itoa(start))
			} else {
				result = append(result, fmt.Sprintf("%d->%d",start,end))
			}            
            
            if i < len(nums) {
				start = nums[i]
			}
        }
    }

    return result;
}

File: intervals/summary-ranges/summary-ranges.ts
function summaryRanges(nums: number[]): string[] {
    const result: string[] = [];

    if (nums.length === 0) return result;

    let start = nums[0];

    for (let i = 1; i <= nums.length; i++)
    {
        if (i === nums.length || nums[i] !== nums[i - 1] + 1)
        {
            const end = nums[i - 1];
            result.push(start === end ? `${start}` : `${start}->${end}`);
            
            if (i < nums.length) start = nums[i];
        }
    }

    return result;
};